bob-backend/src/core.rs

```diff
@@ -260,16 +262,18 @@ impl Backend {
                 self.put_single(key, data, op).await?;
             }
             Ok(())
-        } else if let Some(path) = disk_path {
+        } else if let Some(paths) = disk_paths {
             debug!(
                 "remote nodes is empty, /{:.3}ms/",
                 sw.elapsed().as_secs_f64() * 1000.0
             );
-            let res = self
-                .put_single(key, data, Operation::new_local(vdisk_id, path))
-                .await;
+            for p in paths {
+                if let Err(e) = self.put_single(key, data, Operation::new_local(vdisk_id, p.clone())).await {
+                    warn!("PUT[{}] error put to {:?}: {:?}", key, p, e);
+                }
+            }
             trace!("put single, /{:.3}ms/", sw.elapsed().as_secs_f64() * 1000.0);
-            res
+            Ok(())
```

Comment:
Errors are lost here. We must return an error if the write to at least one replica ended with error

Reply:
In `delete` this is implemented correctly